### ISSUE TITLE
No support for 'using enum'

### ISSUE THREAD

**Issue by DanielKriz** [tree-sitter/tree-sitter-cpp#172]
As of C++20 it is possible to use `using enum <name>;`

The parser does not support it yet.

```cpp
enum class Foo {
    bar,
};
int main()
{
    using enum Foo;
}
```
And `TreeSitterPlayground` shows:
```
enum_specifier [0, 0] - [2, 1]
  name: type_identifier [0, 11] - [0, 14]
  body: enumerator_list [0, 15] - [2, 1]
    enumerator [1, 4] - [1, 7]
      name: identifier [1, 4] - [1, 7]
function_definition [3, 0] - [6, 1]
  type: primitive_type [3, 0] - [3, 3]
  declarator: function_declarator [3, 4] - [3, 10]
    declarator: identifier [3, 4] - [3, 8]
    parameters: parameter_list [3, 8] - [3, 10]
  body: compound_statement [4, 0] - [6, 1]
    using_declaration [5, 4] - [5, 19]
      identifier [5, 10] - [5, 14]
      ERROR [5, 15] - [5, 18]
        identifier [5, 15] - [5, 18]
```
There is an `error` instead of `enumerator` identifier.

**Comment by DanielKriz** [comment]
I don't know how does the treesitter grammar work, but you would point me out. I would like to help resolve this.

**Comment by alfaix** [comment]
Not a maintainer but the documentation for writing tree-sitter grammar is [here](https://tree-sitter.github.io/tree-sitter/creating-parsers).

You only need to change the `grammar.js` and test files (the .txt ones), everything else is generated by tree-sitter.
You probably want to alter [this rule](https://github.com/tree-sitter/tree-sitter-cpp/blob/master/grammar.js#L645).

**Comment by jdrouhard** [comment]
Fixed by 5ead1e26c6ab71919db0f1880c46a278a93bc5ea
